internal/lsp/stdio.go

```diff
@@ -0,0 +1,20 @@
+package lsp
+
+import "os"
+
+type StdOutReadWriteCloser struct{}
+
+func (StdOutReadWriteCloser) Read(p []byte) (int, error) {
+	return os.Stdin.Read(p)
+}
+
+func (c StdOutReadWriteCloser) Write(p []byte) (int, error) {
+	return os.Stdout.Write(p)
+}
+
+func (c StdOutReadWriteCloser) Close() error {
+	if err := os.Stdin.Close(); err != nil {
+		return err
+	}
+	return os.Stdout.Close()
+}
```

Comment:
When is it required to _Close()_ os.Stdin and os.Stdout? Genuinely curious 😅

Reply:
Hmm, this was based on another LSP implementation. Upon further research it seems like it's not a good idea to close stdout or in, so I have removed this and made the function a nop.